src/ol/proj/proj.js

```diff
@@ -120,6 +120,9 @@ ol.proj.Projection = function(options) {
    * @type {boolean}
    */
   this.global_ = goog.isDef(options.global) ? options.global : false;
+  if (this.global_) {
+    goog.asserts.assert(!goog.isNull(this.extent_));
```

Comment:
When using a projection for transforming coordinates only, you don't need an extent - and you also don't need to set isGlobal. I think the best will be to remove the assertion altogether like you suggested. To avoid checking for the same conditions over and over, I can add a canWrapX function (which returns with a cached value) to ol.proj.Projection.